File: DSA2026/maxProfit.py

prices = [6,7,4,2,10,4,14]

def maxProfit(prices):
    r,l = 1,0
    maxP = 0
    
    while r< len(prices):
        if prices[l] < prices[r]:
            profit = prices[r] - prices[l]
            maxP = max(maxP, profit)
        else:
            l = r
        r= r+1
    return maxP

print(maxProfit(prices))

File: DSA2026/maxProfit.ts
const prices: number[] = [7, 1, 5, 3, 6, 4];

function maxProfit(prices: number[]): number {
  let l = 0;
  let r = 1;
  let maxP = 0;

  while (r < prices.length) {
    if (prices[l] < prices[r]) {
      let profit = prices[r] - prices[l];

      maxP = Math.max(profit, maxP);
    } else {
      l = r;
    }
    r++;
  }
  return maxP;
}

console.log(maxProfit(prices));
